akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -230,63 +237,46 @@ private[akka] object StreamLayout {
     override def materializedValueComputation: MaterializedValueNode = Ignore
   }
 
-  final case class CompositeModule(
-    subModules: Set[Module],
-    shape: Shape,
-    override val connections: List[(OutPort, InPort)],
-    override val materializedValueComputation: MaterializedValueNode,
-    attributes: OperationAttributes) extends Module {
+  final case class CopiedModule(shape: Shape, attributes: OperationAttributes, copyOf: Module) extends Module {
+    override val subModules: Set[Module] = Set(copyOf)
+
+    override def withAttributes(attr: OperationAttributes): Module = this.copy(attributes = attr)
+
+    override def carbonCopy: Module = this.copy(shape = shape.deepCopy())
```

Comment:
If I lift this up we can get rid of copy methods on atomic modules. Since I was not sure of all the consequences it might have I left those as they are now. We can revisit it later.